src/Support/ExecutionMiddleware.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types = 1);
+namespace Rebing\GraphQL\Support;
+
+use Closure;
+
+abstract class ExecutionMiddleware
+{
+    /**
+     * @param string|mixed $query
+     * @param array<mixed>|mixed $args
+     * @return Closure|array<mixed>
+     */
+    public function handle($query, $args, Closure $next)
+    {
+        return $next($query, $args);
+    }
+
+    /**
+     * @param array<string,mixed> $arguments
+     * @return Closure|array<mixed>
+     */
+    public function resolve(array $arguments, Closure $next)
+    {
+        return $this->handle(...$arguments, ...[
+            function (...$arguments) use ($next) {
+                return $next($arguments);
+            },
+        ]);
```

Comment:
I don't think I'll ever understand this brainf..kery of code 😛

Reply:
Ah, it "clicked" now; essentially this is this:
```php
        $arguments[] = function (...$arguments) use ($next) {
            return $next($arguments);
        };

        return $this->handle(...$arguments);
```

But once unpacking is used, you can't have positional arguments anymore.

Reply:
haha me neither, just copied: :smile: